File: benchmarks/bench_hash.js
"use strict";
const crypto = require("crypto");
const buf = "the quick brown fox jumps over the lazy dog 0123456789 ".repeat(1000);
const total = 50;
let last = "";
for (let i = 0; i < total; i++) last = crypto.createHash("sha256").update(buf).digest("hex");
if (last.length !== 64) throw new Error("len");
console.log("hash rounds =", total, "digest =", last);

File: benchmarks/bench_hash.go
package main

import (
	"crypto/sha256"
	"encoding/hex"
	"fmt"
	"strings"
)

func main() {
	buf := strings.Repeat("the quick brown fox jumps over the lazy dog 0123456789 ", 1000)
	total := 50
	last := ""
	for i := 0; i < total; i++ {
		h := sha256.Sum256([]byte(buf))
		last = hex.EncodeToString(h[:])
	}
	if len(last) != 64 {
		panic("len")
	}
	fmt.Println("hash rounds =", total, "digest =", last)
}
